cgroup_top.py

```diff
@@ -284,6 +289,31 @@ def init():
             if arg in subsystems and arg not in CGROUP_MOUNTPOINTS:
                 CGROUP_MOUNTPOINTS[arg] = mount[1]
 
+def get_user_beacounts(user):
+    '''
+    get memory stats from vzlist output to openvz containers
+    special user -2 means: current user
+    '''
+    prefix = []
+    cur_uid = os.getuid()
+    try:
+        cur_user = pwd.getpwuid(cur_uid).pw_name
+    except:
+        cur_user = cur_uid
+
+    if user != cur_user and user != -2:
+        if cur_uid == 0:
+            prefix = ['su', user]
+        if user == 'root':
+            prefix = ['sudo']
+        else:
+            prefix = ['sudo', '-u', user]
+
+    command = " ".join(prefix) + " " + "vzlist -o ctid,privvmpages,privvmpages.l -H"
```

Comment:
Use a list instead, for security. This will avoid spawning an intermediate shell. Something like:

```
command = prefix + ['vzlist', '-o', 'ctid,privvmpages,privvmpages.l', '-H']
```